src/app/home/records/records.component.ts

```diff
@@ -14,8 +20,40 @@ export class RecordsComponent implements OnInit {
   userId: string = JSON.parse(localStorage.getItem('user')!).id;
 
   private accountingService: AccountingService = inject(AccountingService);
+  private dialog: MatDialog = inject(MatDialog);
+  private destroyRef: DestroyRef = inject(DestroyRef);
 
   ngOnInit() {
     this.userCategories$ = this.accountingService.getCurrentUserCategories(this.userId);
   }
+
+  saveEvent(formData: FormGroup<EventFormModel>) {
+    const event: EventModel = {
+      id: uuidv4(),
+      type: formData.value.type!,
+      description: formData.value.description!,
+      date: new Date().toLocaleString(),
+      userId: this.userId,
+      amount: Number(formData.value.amount),
+      categoryId: formData.value.categoryId!
+    };
+
+    this.accountingService.createEvent(event).pipe(
+      takeUntilDestroyed(this.destroyRef)
+    ).subscribe();
+  }
+
+  openAddEventModal(): void {
+    this.dialog.open(AddEventModalComponent, {
```

Comment:
you can create config file with this params, bacause I see that you are reising it